### PR TITLE
Reduced radiation contamination spreading and SM abuse

### PR DESCRIPTION
**Pull request by WarlockD**
## About The Pull Request
Reduced the amount of radiation you get from someone who is contaminated by a forth.  So if that CE comes in medical with 2000 mev, at most anyone will get is irradiated by 500.

Oh, and dumping a ton of matter bins in front of the SM for more power, thats fixed to.  Your welcome!

## Why It's Good For The Game
This will stop radiation griefing as well as not dying by just looking at the CE.  Also make the engineers do their job.

## Changelog
:cl:
fix: Reduced the amount of radiation contamination spreading
fix: Fixed the collectors to only eat the highest rad source
add: You can now build washing machines!
tweak: Washing machines now wash contents with bluespace magic!
fix: Sinks and showers properly wash normally and contamination!
/:cl:

### CODE REVIEW

File: code/__DEFINES/radiation.dm [hunk 1]
```diff
@@ -49,4 +49,4 @@ Ask ninjanomnom if they're around
 #define RAD_CONTAMINATION_STR_COEFFICIENT 0.25		// Higher means higher strength scaling contamination strength
 #define RAD_DISTANCE_COEFFICIENT 1					// Lower means further rad spread
 
-#define RAD_HALF_LIFE 90							// The half-life of contaminated objects
+#define RAD_HALF_LIFE 90							// The half-life of contaminated objects in
```
Comment (LemonInTheDark) [inline code/__DEFINES/radiation.dm:52]: bruh

**Comment by Trumpetplaya288** [comment]
making it easier to get rid of rads would be better nerf
even if i have a measly 10k rads i still need to take a 10 minute shower to get rid of it all and i cant decontaminate my suit in the EVA unit because ill get dosed as soon as i take it off, plus i think its bugged anyway

File: code/__DEFINES/radiation.dm (hunk 1 above)
Comment (WarlockD) [reply code/__DEFINES/radiation.dm:52]: Sorry I did a change to the comment and thought I reverted it, just rushed it

**Comment by WarlockD** [comment]
And you will still be in a 10 min shower because you have RADIATION SICKNESS AND THE SHOWER DOES NOTHING FOR THAT.  Any item on your person that is contaminated will be clean in 5 minutes tops, no matter how contaminated they are.  You will still be irradiated and stay that way till you get some meds, you just won't spread it.

Just drop all your stuff into a backpack, throw it into a suit storage unit and decontaminate.  Jump in a shower so you don't spread it then go to medbay for some meds

**Comment by Trumpetplaya288** [comment]
> 
> 
> And you will still be in a 10 min shower because you have RADIATION SICKNESS AND THE SHOWER DOES NOTHING FOR THAT. Any item on your person that is contaminated will be clean in 5 minutes tops, no matter how contaminated they are. You will still be irradiated and stay that way till you get some meds, you just won't spread it.
> 
> Just drop all your stuff into a backpack, throw it into a suit storage unit and decontaminate. Jump in a shower so you don't spread it then go to medbay for some meds

if you're so salty about plasmemes lighting up medbay dont you think that giving civilians a better means to scrub their contamination would be a better solution? changing the spread coefficient might help a little but the BRPED fix isnt going to protect you against fusion

**Comment by WarlockD** [comment]
That's the thing, your engineers, you shouldn't be screwing with fusion without a rad suit.  You wear a rad suit you are 100% protected.  Will your rad suit be contaminated?  YES, so shower if you are leaving engineering.  The radiation storm doesn't even create contamination so the only way a random non-engineer can get it is if they are doing a naughty.  If the station is getting poisoned by radiation before or after this nerf, it's all on the engineers heads.

I am not even salting about plasmemes.  Medical responsibility to know on how to handle the species they have.  Like not throwing lizard people into the tube or taking off a plasmamans suit.

As better way to scrub, I might just increase the shower washing.  At the very least of using a shower you should get clean from it much quicker.  I mean hell, in real decon rooms, soup and water is amazingly effective.

**Comment by ATH1909** [comment]
*fourth

*soap

**Comment by necromanceranne** [comment]
Not sure whether or not it was diagnosed as the actual problem, but discussion in codebus suggested likely large amounts of contamination could be adversely affecting server performance. I'll be honest, I think it'd be better to just kill off contamination in it's entirety.

**Comment by ominousgrace** [comment]
Hello, i applaud your matter bin fix. However i don't think that you should include the upteenth radiation nerf in the same PR.

**Comment by WarlockD** [comment]
Yea it should be two prs but the conversation was all about all the radiation problems and complications and kept using that exploit as the reason there should be a nerf.

**Comment by WarlockD** [comment]
> Not sure whether or not it was diagnosed as the actual problem, but discussion in codebus suggested likely large amounts of contamination could be adversely affecting server performance. I'll be honest, I think it'd be better to just kill off contamination in it's entirety. It's a lame mechanic even if it wasn't causing the server to come to an absolute standstill.

That can be fixed but it's more complicated so need to put it in another pr.  The big cause of the performance is people dumping like 50 folded uranium chair and each one creates a wave.  While I like the idea of waves, the way they are implemented is stupid.  Spawning a datum in 4 directions is stupid.  Its cheaper to give a tile a "radiation" counter that just auto degrades and only changes when a stronger source is produced rather than having 50 pulses at once.

**Comment by Tlaltecuhtli** [comment]
how to remove rads without being an idiot who showers: stick contaminated clothes in the ssu and decontaminate this makes the person not spread any rads  and safe to go anywhere, no need for more rads nerf because you dont know how mechanics work

**Comment by Doosteroni** [comment]
> That's the thing, your engineers, you shouldn't be screwing with fusion without a rad suit. You wear a rad suit you are 100% protected. 

Correct me if I’m wrong but rad suits don’t protect against all sources of radiation

**Comment by LemonInTheDark** [comment]
They should be totally impermeable. I think they may have broken from the red sun glitch, but that and any related issues **should** be fixed. If you have a round id where it hasn't worked, yell at me, I love a good bug.

**Comment by WarlockD** [comment]
> Correct me if I’m wrong but rad suits don’t protect against all sources of radiation
They do.  Even if you have something in your bag or even pocket, the rad search starts from the outermost container, the suit in this case and the suit stops everything.  And the helmet, can't forget that hah.

It doesn't stop the suit itself from getting contaminated.  It just stops everything in the suit from being so.

**Comment by Doosteroni** [comment]
I’ve done multiple shard removals and every time I spend 10 min in med even though I wore a suit. I’ll do some testing and then make an issue or some shit

**Comment by Doosteroni** [comment]
> I’ve done multiple shard removals and every time I spend 10 min in med even though I wore a suit. I’ll do some testing and then make an issue or some shit

As of current it isn't an issue so nevermind

**Comment by WarlockD** [comment]
Not stale but after talking with coding, going to add some machine washing code as well as making the tubes clean as well

**Comment by WarlockD** [comment]
I am digging in some old code now.  If I can finish this patch tommow I think I will work on an acid component  That is some old crap there

**Comment by WarlockD** [comment]
Fun fact, the shower did NOT wash contamination off your suit ugh.

**Comment by stylemistake** [comment]
Walking chernobyls are not realistic, and this makes things slightly more realistic and manageable, so I approve it.

**Comment by WarlockD** [comment]
Ok, I am going to close this and split it up in to a few PR's.  So far I have modifyied the medical kosask, made the washing machine buildable, fixed the sink, the washing mechanic AND working on converting radiation waves to be more of a field.

Way to much:P  Going to split each change in to seperate pr's so I don't go mad

**Pull request closed without merging**